### DISCUSSION TITLE
Google OAuth -- Change displayed redirect url

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [supabase/supabase#2925] [Questions] [answered]
When following the [guide for setting up Google OAuth](https://supabase.io/docs/guides/auth/auth-google), a Supabase url is shown to users upon sign-in instead of my app url. How can you change that?

<img width="479" alt="Screen Shot 2021-08-21 at 4 59 38 PM" src="https://user-images.githubusercontent.com/5659171/130337718-6858d257-7a74-47a9-94a1-ac7b2fa9fd6c.png">

**Comment by silentworks** [comment]
At the moment I don't think this can be changed. But I'm sure in the future you will be able to change this.

**Comment by liaujianjie** [comment, marked as answer]
I just discovered a way around this. If you try logging in to my app ([Mobbin](https://mobbin.design/)) now, you'll see that the name displayed is our app name instead of any kind of domain name:

![CleanShot 2021-10-21 at 16 10 46](https://user-images.githubusercontent.com/3143132/138240154-5193af6e-f505-47bf-a07d-15f6c295a6c7.png)

You can replace the domain name shown on the Google OAuth screen with the _App name_ configured on Google Cloud > APIs and services > OAuth consent screen.

![image](https://user-images.githubusercontent.com/3143132/138239649-0c5f0cab-cc00-4233-8316-891277e29cbf.png)

To do this, you'd need to trigger the OAuth consent screen verification process. I got it triggered by uploading a logo. After triggering the verification process, you should receive an email asking your to verify the ownership of the domains listed on your authorized domains list:

![image](https://user-images.githubusercontent.com/3143132/138240073-e513dd14-9ab0-4a15-824d-40f059f445ee.png)

For your own domain, you'd need to go through the Google domain verification process. For domains that you do not own (e.g. Supabase or other services), you'd need to reply back to an email that you'll receive (I got in only the next day). You should reply stating that those belong to 3rd party services which you use to integrate with Google OAuth.

Note that you'll also need to provide a privacy policy and ToS link during verification.

**Reply by zakaria-chahboun** [reply]
@liaujianjie thanks!
The question should be closed now

**Reply by tourbillonlabs** [reply]
For reference. I am in the process of doing this same thing and it's a bit like pulling teeth. Despite confirming that supabase is my third party auth integration provider and sharing links to tickets on supabase where other users have managed to get their supabase auth domain verified, I keep receiving seemingly canned responses back with variations of:

"Unfortunately, it is against Google API User Data Policy for multiple unique domains associated with distinct brands to be verified under the same project."

I am persisting and linking the relevant comments, but the process is not a seamless one.

**Reply by fabpicapp** [reply]
Yeah same thing they are doing with me as well.
I don't know how those people are in google.
Can't they understand this simple thing and keep on repeating same thing.
I am still not able to get verified.

**Comment by tourbillonlabs** [comment]
Aloha,

I've have just successfully had my app verified. Google did not care a whit that Supabase is my third party authentication provider, and irrespective of what I said, sent back a canned response informing me I was: "in direct violation of the Google API Services: User Data Policy, which requires that projects accurately represent their identity and intent to Google and to our users when they request access to Google user data." and provided instructions to move the offending domains into another project (18 emails in total).

For those who are also on this path, this is what worked for me. I only had non-sensitive scopes selected.

When asked to confirm ownership of the domains, you will see your domain and the Supabase domain listed. Confirm ownership of your domain as they instruct, but in the oAuth Consent Screen, under Authorised Domains, ensure only your owned domains are in the list. If you see Supabase, **clear the field before you click the trash button for the line item**. I don't recall adding this in myself, but I cannot confirm if it was automatically added or not.

As long as Supabase is still in your Authorised redirect URIs in the OAuth 2.0 Client IDs, auth should still work as expected (confirm this once you remove the Authorised Domain line item). Then respond advising your domains are confirmed, but that Supabase is still in the OAuth 2.0 Client IDs because they're your third party authentication provider. They don't acknowledge other than to approve or send a canned response, but it's probably worth mentioning.

Should you get past the domain ownership hurdle, the only additional step required for me was to add more detail to my privacy policy to indicate what Google oAuth info was used for.

I hope this helps others.

**Reply by fabpicapp** [reply]
Once I try to remove the supabse domain from the listed Authorised domains it asks me to delete from the client URIs and when I delete the client redirect URI the login doesn't work.
![screenshot-console cloud google com-2022 01 07-19_01_02](https://user-images.githubusercontent.com/96480786/148551208-b17853ab-53b1-499f-acbe-0ad1687970d6.png)

**Reply by AdrianKrebs** [reply]
@fabpicapp how did you solve this issue? I'm not able to remove the supabase URI either.

**Reply by eMeRiKa13** [reply]
Yes same for me...

**Reply by fabpicapp** [reply]
Hey @AdrianKrebs  and @eMeRiKa13 .
Actually you can't remove this supabase URL.
You just have to tell them in the mail that this domain belongs to the 3rd party Supabase I am using to impliment my oAuth.
Though they will not listen in one mail but after a few to and fro of emails finally they will approve it.
For me after 11 emails of to and fro they approved.
Just keep saying that this is the 3rd Party domain.
[Here](https://github.com/supabase/supabase/discussions/2925#discussioncomment-1945427) I have mentioned it.

**Reply by tourbillonlabs** [reply]
Empty the field and then hit trash on the line item. You can't delete it when the field is populated.

**Reply by tourbillonlabs** [reply]
I've had a second app approved using this method.

The only additional step required was that they said my Sign in with Google button didn't match their style guide. Updated it and it went through immediately.

**Comment by fabpicapp** [comment]
After a lot of to and fro(11 emails in total) they finally approved my project. I had to mention this again and again the supabase link is my 3rd part oAuth provider. And my google signin button was not as per their guideline so I had to make the change for it. Now if you login to my [app](https://fabpic.app/) it will show you my app logo and name.

**Reply by stretch07** [reply]
@fabpicapp having the same issue, would you be able to share their support email? I haven't been able to find it...

**Reply by miketromba** [reply]
> @fabpicapp having the same issue, would you be able to share their support email? I haven't been able to find it...

@its-pablo Once you submit your app for verification in the dashboard, the Google "Trust and Safety" team will send you an email to start the verification process. You can go back and forth with them there to resolve any issues.

See the "Submitting your app for verification" section: 
https://support.google.com/cloud/answer/9110914?hl=en#zippy=%2Csteps-to-submit-your-app

**Comment by miketromba** [comment]
Going through this process now.

It would be great if Supabase added a feature where we could map a domain of our own (e.g. supabase.mydomain.io) to the supabase project domain (e.g. arbitrarychars.supabase.co) using a CNAME DNS record.

This would help us avoid this messy verification process since our callback URLs would be through our own domain.

**Reply by miketromba** [reply]
Update: Took a few days of back and forth but they eventually approved my app. Just be clear with them that the Supabase domain is a 3rd-party backend as a service provider and they should understand.

**Comment by eMeRiKa13** [comment]
I just did the verification process, I explained at the beginning that Supabase is a third party backend and authentication provider. I had no questions about it, and my application was approved!

Maybe the Google validation people know better Supabase now.

**Comment by IshanAmin** [comment]
How do you do the verification process? Is there a form or email?

**Comment by shaon72** [comment]
@liaujianjie Hey Liau, do you know how the verification process can be triggered now?

**Comment by mrkhaniftikhar** [comment]
Hi, just to help and keep this thread updated. With the help of this discussion, we were finally able to remove the Supabase project code. The good news for newcomers is that Google did not send any email — the process was automatic, and our brand got verified in less than 15 minutes. Now it shows our brand name Werkal ( which is AI Resume Builder and Optimizer) along with our logo. I’ve attached a snapshot for reference.

Basically, there are a few simple steps:

1. Upload your logo (if you haven’t already).
2. Add your homepage, Privacy Policy, and Terms of Service links.
3. Most important: In the Authorized domains section, add your main domain first, and Supabase (or any other domain) second.
4. Click Save.
5. You will receive a verification notification inside the Google Cloud Console.
6. Go to Google Search Console and add your domain.
7. Follow the instructions — add the provided TXT record in your domain provider settings.
8. Return to Google Console, wait 5–10 minutes, and click Verify.
9. Change the app status from Testing to Published (if applicable).
10. You should then see the confirmation message at the top, as shown in the circled area in the screenshot.


<img width="1157" height="943" alt="image" src="https://github.com/user-attachments/assets/d707517b-36af-4031-bce0-1c143e1cb3ff" />

**Reply by toadaly** [reply]
A while later, but this helped me out, thanks!

**Reply by IshanAmin** [reply]
Well...Initially, I forgot what I even asked. It's been so long. LOL 
but...
I still needed this. So I'm super appreciative of you providing this much detail in logical steps 

Thanks for doing it.
